### PR TITLE
DC-1907: creating admin pages for whitelist management

### PR DESCRIPTION
**Pull request by rwalpole**

### CODE REVIEW

File: test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      private val fakeRequestWithSession = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      val fakeRequestWithBody: FakeRequest[AnyContentAsJson] = fakeRequestWithSession.withJsonBody(
+        Json.parse(
+          """
+            |{
+            | "blah" : "blah"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.addFormIdToWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
```
Comment (elabeca) [inline test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala]: Not sure that mock is required, but I could be wrong

File: test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+      private val fakeRequestWithSession = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      val fakeRequestWithBody: FakeRequest[AnyContentAsJson] = fakeRequestWithSession.withJsonBody(
+        Json.parse(
+          """
+            |{
+            | "blah" : "blah"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.deleteFormIdFromWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
```
Comment (elabeca) [inline test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala]: Same here, think maybe the controller would throw the error before it reaches the connector. Maybe not needed?

Comment (rwalpole) [reply test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala]: Yep, you're right @elabeca. Now removed.

File: test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala (hunk 1 above)
Comment (rwalpole) [reply test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala]: Yep, you're right @elabeca. Now removed.

**Pull request merged**